opensearch_py_ml/ml_commons/ml_commons_client.py

```diff
@@ -194,6 +195,36 @@ def _get_task_info(self, task_id: str):
             url=API_URL,
         )
 
+    def search_task(self, **kwargs) -> object:
+        """
+        This method searches a task from opensearch cluster (using ml commons api)
+        :param kwargs: key value pairs of search task parameters
+        :type kwargs: dict
+        :return: returns a json object, with detailed information about the searched task
+        :rtype: object
+        """
+
+        API_URL = f"{ML_BASE_URI}/tasks/_search"
+
+        query_bool = {"query": {"bool": {"filter": []}}}
+
+        API_BODY = query_bool
+
+        for key, value in kwargs.items():
+            if key == "json":
```

Comment:
are we expecting somebody to mention "json" in their provided value? Can you give example for this what are you expecting customer to provide?

Reply:
Yes, When calling function if user provides named parameter like this: search_task(json={...})

Reply:
How about we make this more intuitive rather than asking customer to provide `json`. Can we do something like this:

```
def my_function(**kwargs):
    for key, value in kwargs.items():
        if isinstance(value, str):
            try:
                value = json.loads(value.replace("'", '"'))
                kwargs[key] = value
            except json.JSONDecodeError:
                pass
        print(f"{key}: {value}")
```

You need to change the code based on your requirement. But what I meant is, if we see somebody is providing a string, then we can try to load that in a json format.

Reply:
Ok, I understood. I didn't take into account the convenience. I will change the code.

Reply:
I believe that I already did it. It's in else block. Providing json is just one option of setting API_BODY. User can also provide just key and according values. Finally, I used json.dumps for formatting, if in example user provided single quote instead of double quote.

Reply:
and also we should throw `except json.JSONDecodeError:` if the json format is not right. before sending the request to api endpoint. 

When you make the changes, please also describe in the PR description what are the ways (what did you send in `kwargs`) you tried to search using your function.

Reply:
Ok, I will add code for checking json.

In the description of PR, I mentioned all ways of providing parameters to function. Sorry if it is not so much descriptive.

Reply:
I meant in your testing for the functionality what parameters you provided to test.  List down all of them, then I can suggest may be other ways to test the functionality.

Reply:
Ok, I got it. Thank you.